endaq/device/linux.py

```diff
@@ -155,23 +155,23 @@ def readRecorderClock(clockFile: Filename,
 def getDeviceList(types: dict, strict: bool = True) -> list:
     """ Get a list of data recorders, as their respective mount points.
     """
-
     result = set()
-
-    for device, mountpoint, fstype, opts, maxfile, maxpath in psutil.disk_partitions():
+    for device in psutil.disk_partitions():
         try:
-            if not os.path.exists(device):
+            if not os.path.exists(device.mountpoint):
+                continue
+            # Skip devices we've already found (Linux oddity workaround)
+            if device.mountpoint in result:
                 continue
             for t in types:
-                if t.isRecorder(mountpoint, strict=strict):
-                    result.add(mountpoint)
+                if t.isRecorder(device.mountpoint, strict=strict):
+                    result.add(device.mountpoint)
```

Comment:
It's a preexisting issue, but could you insert a `break` after adding the mount point to `result`? The same mount point is getting added 2-3x.